sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -71,28 +76,112 @@ def _need_new_token(self):
         return not self._token or self._token.expires_on - time.time() < 300
 
 
-class BearerTokenCredentialPolicy(_BearerTokenCredentialPolicyBase, SansIOHTTPPolicy):
+class BearerTokenCredentialPolicy(_BearerTokenCredentialPolicyBase, HTTPPolicy):
```

Comment:
Is this a breaking change?

Reply:
isinstance(BearerTokenCredentialPolicy, SansIOHTTPPolicy) will change from True to False?

Reply:
The contract didn't change, Python is more about protocol that types, so if the class behave the same I'm good (which doesn't seem to be the case, see my other comment)

Reply:
My concern is code like

```py
for policy in policies:
      if isinstance(policy, SansIOHTTPPolicy):
              do_sth()
```

e.g. https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure-core/azure/core/pipeline/_base.py#L133